pkg/expr/sql/frame_db.go

```diff
@@ -0,0 +1,63 @@
+package sql
+
+import (
+	mysql "github.com/dolthub/go-mysql-server/sql"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+var dbName = "frames"
+
+// FramesDBProvider is a go-mysql-server DatabaseProvider that provides access to a set of Frames.
+type FramesDBProvider struct {
+	db mysql.Database
+}
+
+func (p *FramesDBProvider) Database(_ *mysql.Context, _ string) (mysql.Database, error) {
+	return p.db, nil
+}
+
+func (p *FramesDBProvider) HasDatabase(_ *mysql.Context, _ string) bool {
+	return true
+}
+
+func (p *FramesDBProvider) AllDatabases(_ *mysql.Context) []mysql.Database {
+	return []mysql.Database{p.db}
+}
+
+// NewFramesDBProvider creates a new FramesDBProvider with the given set of Frames.
+func NewFramesDBProvider(frames data.Frames) mysql.DatabaseProvider {
+	fMap := make(map[string]mysql.Table, len(frames))
+	for _, frame := range frames {
+		fMap[frame.RefID] = &FrameTable{Frame: frame}
+	}
+	return &FramesDBProvider{
+		db: &framesDB{
+			frames: fMap,
+		},
+	}
+}
+
+// framesDB is a go-mysql-server Database that provides access to a set of Frames.
+type framesDB struct {
+	frames map[string]mysql.Table
+}
+
+func (db *framesDB) GetTableInsensitive(_ *mysql.Context, tblName string) (mysql.Table, bool, error) {
```

Comment:
🤔 ref ID is case-sensitive. is this going to be an issue here?
<img width="742" alt="image" src="https://github.com/user-attachments/assets/3f85020a-bdb2-4ffe-8c49-4102fb095676" />

Reply:
Yes, may fix post this PR though - since we are still in experimental.